File: examples/python/basic.py
"""
Fortune Cookie Generator API - Basic Usage Example

This example demonstrates the basic usage of the Fortune Cookie Generator API.
API Documentation: https://docs.apiverve.com/ref/fortunecookie
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/fortunecookie'

def call_fortunecookie_api():
    """
    Make a GET request to the Fortune Cookie Generator API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;count&#x27;: 3, &#x27;type&#x27;: &#x27;wisdom&#x27;}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Fortune Cookie Generator API...\n')

    result = call_fortunecookie_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: npm/examples/basic.js
/**
 * Basic Example - Fortune Cookie Generator API
 *
 * This example demonstrates how to use the Fortune Cookie Generator API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const fortunecookieAPI = require('../index.js');

// Initialize the API client
const api = new fortunecookieAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  count: 3,
  type: "wisdom"
};

// Make the API request using callback
console.log('Making request to Fortune Cookie Generator API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});
